tests/test_containerproperties.py

```diff
@@ -632,3 +659,91 @@ def test_update_xml_value_with_missing_sub_element(self):
+        qname1 = utils.random_qname(localname=self.sub_element_name.text)
+        qname2 = utils.random_qname(localname=self.sub_element_name.text)
+        setattr(self.instance, self.property._local_var_name, [qname1, qname2])
+        self.property.update_xml_value(self.instance, self.node)
+        sub_node = self.node.find(self.sub_element_name.text, namespaces=self.node.nsmap)
+        self.assertIsNotNone(sub_node)
+        self.assertEqual(
+            sub_node.text,
+            f'{docname_from_qname(qname1, ns_map=self.node.nsmap)} '
+            f'{docname_from_qname(qname2, ns_map=self.node.nsmap)}',
+        )
+
+    def test_update_xml_value_with_mandatory_check(self):
+        """Test when the value is None and the property is mandatory."""
+        self.property._is_optional = False
+        setattr(self.instance, self.property._local_var_name, None)
+        with self.assertRaises(ValueError, msg=f'mandatory value {self.sub_element_name} missing'):
+            self.property.update_xml_value(self.instance, self.node)
+
+    def test_update_xml_value_overwrites_existing(self):
+        """Test that the method overwrites existing sub-element text."""
+        existing_sub_node = etree.SubElement(self.node, self.sub_element_name.localname, nsmap=self.node.nsmap)
+        existing_sub_node.text = docname_from_qname(
+            utils.random_qname(localname=self.sub_element_name.text),
+            ns_map=self.node.nsmap,
+        )
+        qname = utils.random_qname(localname=self.sub_element_name.text)
+        setattr(self.instance, self.property._local_var_name, [qname])
+        self.property.update_xml_value(self.instance, self.node)
+        sub_node = self.node.find(self.sub_element_name.text, namespaces=self.node.nsmap)
+        self.assertEqual(sub_node.text, docname_from_qname(qname, ns_map=self.node.nsmap))
+
+    def test_update_xml_value_creates_sub_element(self):
+        """Test that the method creates a sub-element if it doesn't exist."""
+        qname = utils.random_qname(localname=self.sub_element_name.text)
+        setattr(self.instance, self.property._local_var_name, [qname])
+        self.property.update_xml_value(self.instance, self.node)
+        sub_node = self.node.find(self.sub_element_name.text, namespaces=self.node.nsmap)
+        self.assertIsNotNone(sub_node)
+        self.assertEqual(sub_node.text, docname_from_qname(qname, ns_map=self.node.nsmap))
+
+
+class TestAttributeListBase(unittest.TestCase):
+    def setUp(self):
+        self.q_name = utils.random_qname()
+        self.attribute_name = utils.random_qname_part()
+        value_converter = mock.MagicMock()
+        value_converter.elem_to_xml = str
+        self.property = _AttributeListBase(
+            attribute_name=self.attribute_name,
+            value_converter=value_converter,
+        )
+        self.instance = mock.MagicMock()
+        self.node = etree.Element('Root', nsmap={self.q_name.localname: self.q_name.namespace})

```

Comment:
misusage of self.q_name.localname in nsmap - nsmap here can be removed.
This also applies for other occurrences in this PR, whereas some unittest have issues when removing the nsmap. In my opinion the usage of e.g.
`utils.random_qname(localname=self.sub_element_name.text)` is somehow wrong, which causes failing unittest when removing or adapting the nsmap.

Reply:
done